File: Day-03/35_Search_Insert_Position.java
import java.util.Arrays;
class Anshu{
    static int position(int test_case[], int target){
        int index;
        int start = 0;
        int end = test_case.length-1;
        while(start<end){
            int mid = start + (end-start)/2;
            if (test_case[mid] == target){
                return mid;
            }if(test_case[mid]>target){
                end = mid-1;
            }else{
                start = mid+1;
            }
        }
        return start+1;
    }
    public static void main(String[] args) {
        int test_case_1[] = new int[]{1,3,5,6};
		int target_1 = 5;
		int test_case_2[] = new int[]{1,3,5,6};
		int target_2 = 2;
		int test_case_3[] = new int[]{1,3,5,6};
		int target_3 = 7;

        System.out.println("--- test case 1 ---");
		System.out.println("Input: nums = "+Arrays.toString(test_case_1)+" , target = "+target_1+"\nOutput: "+position(test_case_1,target_1));
		System.out.println("--- test case 3 ---");
		System.out.println("Input: nums = "+Arrays.toString(test_case_2)+" , terget = "+target_2+"\nOutput: "+position(test_case_2,target_2));
		System.out.println("--- test case 3 ---");
		System.out.println("Input: nums = "+Arrays.toString(test_case_3)+" , terget = "+target_3+"\nOutput: "+position(test_case_3,target_3));
	

    }
}

File: Day-03/35_Search_Insert_Position.py
def position(test_case,target):
    
    start = 0
    end = len(test_case)-1
    while(start<end):
        mid = int(start + (end-start)/2)
        if (test_case[mid] == target):
            return mid
        if(test_case[mid] > target):
            end = mid-1
        else:
            start = mid+1
    return start+1


test_case_1 = [1,3,5,6]
target_1 = 5
test_case_2 = [1,3,5,6]
target_2 = 2
test_case_3 = [1,3,5,6]
target_3 = 7

print("--- test case 1 ---")
print(f"Input: nums = {test_case_1} , target = {target_1}\nOutput: {position(test_case_1,target_1)}")
print("--- test case 2 ---")
print(f"Input: nums = {test_case_2} , target = {target_2}\nOutput: {position(test_case_2,target_2)}")
print("--- test case 3 ---")
print(f"Input: nums = {test_case_3} , target = {target_3}\nOutput: {position(test_case_3,target_3)}")